File: js/src/ArrayProblem/TwoSum.js


export function TwoSum(nums, target) {
    const bag = {};

    for (let i = 0; i < nums.length; i++) {
        let comp = target - nums[i];
        if (comp in bag) {
            return [bag[comp], i];
        } else {
            bag[nums[i]] = i;
        }
    }
    return null; 
}

File: python/two_pointer/two_sum.py



def get_two_number(nums, target):

    right = len(nums) - 1
    left = 0

    while left <= right:

        total = nums[right] + nums[left]

        if total == target :
            return [left+1 , right+1]
        
        if total > target :
            right -= 1
        else:
            left += 1
    
    return []